File: solutions/go/collatz-conjecture/1/collatz_conjecture.go
package collatzconjecture

import "errors"

func CollatzConjecture(n int) (int, error) {
    t:=0
    if n==1 {
        t=0
        return t , nil
    }
    if n <= 0 {
    return 0, errors.New("number must be positive")
    }else{
		for n!=1 {
        	if n%2 == 0{
            	n=n/2
            	t=t+1
        	}else {
            	n= (n*3)+1
            	t=t+1
        	}
    	}
	}
    return t , nil
}

File: solutions/python/collatz-conjecture/1/collatz_conjecture.py
def steps(number):
    s=0
    if number==1:
        return 0
    elif number<1:
        raise ValueError("Only positive integers are allowed")
    else:
        while number!=1:
            if number%2==0:
                number=number/2
                s+=1
            else:
                number=(number*3)+1
                s+=1
    return s